service/header/store_heightsub.go

```diff
@@ -25,6 +25,11 @@ func (hs *heightSub) Height() uint64 {
 	return atomic.LoadUint64(&hs.height)
 }
 
+// SetHeight sets the new head height for heightSub.
+func (hs *heightSub) SetHeight(height uint64) {
```

Comment:
Does this need to be called from anywhere outside the header pkg?

Reply:
No, same for Pub and Sub and Height.

Reply:
Even though their names are public, those methods will never be accessed outside the pkg, as the constructor is private. 
I keep methods capitalized even for structs in the same pkg. This ways it's clearer that some code call the method from a dependency structure.

Reply:
I mean the only time I ever see private structs with exported methods are when struct is implementing an interface. Is it really necessary to export them when this isn't the case?

Reply:
It's not necessary but it is a code style we have throughout the project. IMO, it makes things clearer. The rule is: a structure that is meant to be used privately inside a pkg, should have capitalized methods if those methods are meant to be used by other structures within the same pkg. Non-capitalized methods of a structure are meant to be used only within the structure.

Reply:
Agh i really don't like `SetHeight` being exported. I know it's not reachable but still.

Reply:
Why specifically SetHeight and not others though. Is this because it is a setter that feels public while it is not?

Reply:
Yes.